packages/rum-core/src/domain/configuration.ts

```diff
@@ -144,6 +146,109 @@ export function validateAndBuildRumConfiguration(
+    if (!Array.isArray(initConfiguration.configureTracingUrls)) {
+      display.error('Configure Tracing URLs should be an array')
+      return
+    }
+    if (initConfiguration.configureTracingUrls.length !== 0 && initConfiguration.service === undefined) {
+      display.error('Service needs to be configured when tracing is enabled')
+      return
+    }
+
+    // Convert from MatchOption | TracingOption to TracingOption
+    const configureTracingUrls: ConfigureTracingOption[] = []
+    for (const option of initConfiguration.configureTracingUrls) {
+      configureTracingUrls.push(
+        isMatchOption(option) ? ({ match: option, headerTypes: ['dd'] } as ConfigureTracingOption) : option
+      )
+    }
+    return configureTracingUrls
+  }
+
+  // Handle conversion of allowedTracingOrigins to configureTracingUrls
+  if (initConfiguration.allowedTracingOrigins !== undefined) {
+    if (!Array.isArray(initConfiguration.allowedTracingOrigins)) {
+      display.error('Allowed Tracing Origins should be an array')
+      return
+    }
+    if (initConfiguration.allowedTracingOrigins.length !== 0 && initConfiguration.service === undefined) {
+      display.error('Service needs to be configured when tracing is enabled')
+      return
+    }
+
+    return initConfiguration.allowedTracingOrigins.reduce((configArray, item) => {
+      const option = convertLegacyMatchOptionToTracingOption(item)
+      if (option) {
+        configArray.push(option)
+      }
+      return configArray
+    }, [] as ConfigureTracingOption[])
+  }
+
+  return []
+}
+
+/**
+ * Converts parameters from the deprecated allowedTracingOrigins
+ * to configureTracingUrls. Handles the change from origin to full URLs.
+ */
+function convertLegacyMatchOptionToTracingOption(item: MatchOption) {
+  let match: MatchOption | undefined
+  if (typeof item === 'string') {
+    match = item
+  }
+  if (item instanceof RegExp) {
+    match = (url) => item.test(getOrigin(url))
+  }
+  if (typeof item === 'function') {
+    match = (url) => item(getOrigin(url))
+  }
+
+  if (match === undefined) {
+    display.warn('Allowed Tracing Origins parameters should be a string, RegExp or function. Ignoring parameter', item)
+    return undefined
+  }
+
+  return { match, headerTypes: ['dd'] } as ConfigureTracingOption
+}
+
+/**
+ * Combines the selected tracing headers from the different options in configureTracingUrls,
+ * and assumes 'dd' has been selected when using allowedTracingOrigins
+ */
+function getSelectedTracingHeaders(configuration: RumInitConfiguration): TracingHeadersType[] {
+  const usedTracingHeaders = new Set<TracingHeadersType>()
+
+  if (Array.isArray(configuration.configureTracingUrls) && configuration.configureTracingUrls.length > 0) {
+    configuration.configureTracingUrls.forEach((config) => {
+      if (isMatchOption(config)) {
+        usedTracingHeaders.add('dd')
+      } else if (Array.isArray(config.headerTypes)) {
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** This `if` seems unnecessary, maybe remove it? If you fear that customers might not define `headerTypes` as an array, you should test this during `validateAndBuildRumConfiguration`.

Reply:
We're working with the InitConfiguration, not the validated configuration here, so there's a small risk.